internal/pkg/users/users.go

```diff
@@ -17,25 +17,61 @@ limitations under the License.
 package users
 
 import (
+	"bytes"
 	"errors"
+	"fmt"
 	"os/exec"
 	"os/user"
 	"strconv"
-	"strings"
 )
 
-// GetUID returns uid of given username and logs a warning if its missing
-func GetUID(name string) (int, error) {
-	entry, err := user.Lookup(name)
-	if err == nil {
-		return strconv.Atoi(entry.Uid)
-	}
-	if errors.Is(err, user.UnknownUserError(name)) {
+const (
+	// An unknown (i.e. invalid) user ID. It's distinct from a UID's zero value,
+	// which happens to be [RootUID]. Assuming root may or may not be a good
+	// default, depending on the use case. Setting file ownership to root is a
+	// restrictive and safe default, running programs as root is not. Therefore,
+	// this is the preferred return value for UIDs in case of error; callers
+	// must then explicitly decide on the fallback instead of accidentally
+	// assuming root.
+	UnknownUID = -1
+
+	RootUID = 0 // User ID of the root user
+)
+
+// Lookup looks up a user's UID by username. If the user cannot be found, the
+// returned error is of type [user.UnknownUserError]. If an error is returned,
+// the returned UID will be [UnknownUID].
+func LookupUID(name string) (int, error) {
+	var uid string
+
+	if entry, err := user.Lookup(name); err != nil {
+		if !errors.Is(err, user.UnknownUserError(name)) {
```

Comment:
`errors.Is(err, user.UnknownUserError(name))` looks weird but seems to work.

Maybe returning a locally defined `UnknownUserError = errors.New("unknown user")` would make the function more pleasant to use.

https://go.dev/play/p/EsG_Q7Uagbw

Reply:
I was also pondering about this. The original code re-used this standard library error in this way. Changing the returned error would trickle down to callers, which are relying on this as well.

What about having a function for this instead, e.g.

```go
func IsNotExist(err error) bool {
	var UnknownUserError user.UnknownUserError
	return errors.As(err, &UnknownUserError)
}
```

Callers could then use `if users.IsNotExist(err) { ... }` ... Which would mimic the design of the standard library's os.Is... functions.

Reply:
Sounds reasonable. The `os.Is...` function docs now do say `new code should use errors.Is(err, fs.ErrNotExist)` but they're not going anywhere.

Reply:
Oh. Right. I'll have a look at your initial proposal first, then.

Reply:
PTAL